### DISCUSSION TITLE
Color change after convert cmyk to srgb?

### DISCUSSION THREAD

**Discussion by zhifengzhuang** [libvips/libvips#3480] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/13492988/236374513-ddade248-2310-495c-a39f-a4573fa25b69.png)
version: macos vips-8.14.2
`vips icc_transform f1137b0a-f124-4372-ae4c-b470c8638d43.jpeg srgb.jpeg srgb`
![image](https://user-images.githubusercontent.com/13492988/236374537-a7782b1f-d460-41fb-8d8d-375c220dea29.png)

**Comment by kleisauke** [comment]
I could not reproduce this with lcms 2.15 and libvips 8.14.2.
```console
$ curl -LO https://user-images.githubusercontent.com/13492988/236374513-ddade248-2310-495c-a39f-a4573fa25b69.png
$ vips icc_transform 236374513-ddade248-2310-495c-a39f-a4573fa25b69.png srgb.png srgb
```
![srgb](https://user-images.githubusercontent.com/12746591/236412765-712cb9b2-2230-40ee-87ab-43133da4eaf8.png)

Homebrew provides lcms 2.14.
https://github.com/Homebrew/homebrew-core/blob/20fc3caac5168fc6f2b78e66177bd49a5f988811/Formula/little-cms2.rb#L6

Someone with access to a macOS machine needs to investigate. I'll see if I can reproduce this on Windows by compiling against lcms 2.14.

**Reply by zhifengzhuang** [reply]
> I could not reproduce this with lcms 2.15 and libvips 8.14.2.
> 
> ```
> $ curl -LO https://user-images.githubusercontent.com/13492988/236374513-ddade248-2310-495c-a39f-a4573fa25b69.png
> $ vips icc_transform 236374513-ddade248-2310-495c-a39f-a4573fa25b69.png srgb.png srgb
> ```
> 
> ![srgb](https://user-images.githubusercontent.com/12746591/236412765-712cb9b2-2230-40ee-87ab-43133da4eaf8.png)
> 
> Homebrew provides lcms 2.14. https://github.com/Homebrew/homebrew-core/blob/20fc3caac5168fc6f2b78e66177bd49a5f988811/Formula/little-cms2.rb#L6
> 
> Someone with access to a macOS machine needs to investigate. I'll see if I can reproduce this on Windows by compiling against lcms 2.14.

Sorry, jpeg converted to png after upload.Here is the original cmyk jpeg url:
```shell
wget http://test-1252081001.cos.ap-shanghai.myqcloud.com/galerie-go/qh/cf17fe98-a65e-4c06-824a-36fc71889c44.jpeg
vips icc_transform cf17fe98-a65e-4c06-824a-36fc71889c44.jpeg cf17fe98-a65e-4c06-824a-36fc71889c44-srgb.jpeg srgb
```

**Reply by zhifengzhuang** [reply]
I also find that cmyk image shows different result in chrome and safari.Vips convert result is more similarity to safari.

**Comment by kleisauke** [comment, marked as answer]
The provided image does not have a embedded CMYK profile, so transforming with `vips_icc_transform()` / `vips_colourspace()` will use the generic CMYK fallback profile, causing C100%+Y100% (`#00FF00`) to be interpreted as `#008E41`.
```python
import pyvips

# C100%+Y100%
green = (pyvips.Image.black(1, 1) + [255, 0, 255, 0]).cast("uchar").copy(interpretation="cmyk")

# [0.0, 142.0, 65.0]
green = green.colourspace("srgb")
print(green(0, 0))
```

Some browsers and viewers may convert CMYK to sRGB via an approximate conversion when the image does not have an embedded CMYK profile. See for example:
https://graphicdesign.stackexchange.com/questions/160359/issue-with-color-profile-of-cmyk-image

This is probably not the proper color you will get from a printer; conversion using a fallback profile should yield much better results.

**Reply by zhifengzhuang** [reply]
thanks for answering, @kleisauke
